tests/common/dualtor/tunnel_traffic_utils.py

```diff
@@ -72,15 +71,77 @@ def _disassemble_ip_tos(tos):
             outer_tos, inner_tos = packet[IP].tos, packet[IP].payload[IP].tos
             outer_dscp, outer_ecn = _disassemble_ip_tos(outer_tos)
             inner_dscp, inner_ecn = _disassemble_ip_tos(inner_tos)
-            logging.debug("Outer packet DSCP: {0:06b}, inner packet DSCP: {1:06b}".format(outer_dscp, inner_dscp))
-            logging.debug("Outer packet ECN: {0:02b}, inner packet ECN: {0:02b}".format(outer_ecn, inner_ecn))
+            logging.info("Outer packet DSCP: {0:06b}, inner packet DSCP: {1:06b}".format(outer_dscp, inner_dscp))
+            logging.info("Outer packet ECN: {0:02b}, inner packet ECN: {0:02b}".format(outer_ecn, inner_ecn))
             check_res = []
             if outer_dscp != inner_dscp:
                 check_res.append("outer packet DSCP not same as inner packet DSCP")
             if outer_ecn != inner_ecn:
                 check_res.append("outer packet ECN not same as inner packet ECN")
             return " ,".join(check_res)
 
+        @staticmethod
+        def _check_queue(self, packet):
+            """Check queue for encap packet."""
+
+            def _disassemble_ip_tos(tos):
+                return tos >> 2, tos & 0x3
+
+            def derive_queue_id_from_dscp(dscp):
+                """ Derive queue id form DSCP using following mapping
+                      DSCP -> Queue mapping
+                        8          0
+                        5          2
+                        3          3
+                        4          4
+                        46         5
+                        48         6
+                        Rest       1
+                """
+
+                if dscp == 8:
+                    queue = 0
+                elif dscp == 5:
+                    queue = 2
+                elif dscp == 3:
+                    queue = 3
+                elif dscp == 4:
+                    queue = 4
+                elif dscp == 46:
+                    queue = 5
+                elif dscp == 48:
+                    queue = 6
+                else:
+                    queue = 1
+
+                return queue
+
+            outer_tos, inner_tos = packet[IP].tos, packet[IP].payload[IP].tos
+            outer_dscp, outer_ecn = _disassemble_ip_tos(outer_tos)
+            inner_dscp, inner_ecn = _disassemble_ip_tos(inner_tos)
+            logging.info("Outer packet DSCP: {0:06b}, inner packet DSCP: {1:06b}".format(outer_dscp, inner_dscp))
+            check_res = []
+            if outer_dscp != inner_dscp:
+                check_res.append("outer packet DSCP not same as inner packet DSCP")
+            exp_queue = derive_queue_id_from_dscp(outer_dscp)
+
+            time.sleep(10)
```

Comment:
why 10sec?. Do you need to clear counters before start? Meaning, will it have some stale values?

Reply:
Added logic to clear queue counters before starting the test.